ledger-tool/src/main.rs

```diff
@@ -2505,17 +2509,21 @@ fn main() {
                             },
                         );
 
-                        let mut unchanged_accounts = stake_calcuration_details
-                            .keys()
-                            .collect::<HashSet<_>>()
+                        // Ensure DashMap lock is dropped
+                        let unchanged_accounts = stake_calculation_details
+                            .iter()
+                            .map(|v| *v.key())
+                            .collect::<HashSet<Pubkey>>();
+
+                        let mut unchanged_accounts = unchanged_accounts
```

Comment:
nit: `// Ensure DashMap lock is dropped by copying the actual inner pubkeys and account data out of it`